frontend/src/components/SearchSettings.js

```diff
@@ -37,7 +37,11 @@ const SearchSettings = () => {
     },
   };
 
+  const [field_to_update, setField] = useState("");
+  const [field_options, setOptions] = useState([]);
   const [settings, setSettings] = useState({});
+  const [open, setOpen] = useState(false);
+  const [save, setSave] = useState(false);
```

Comment:
How do you use those states? What are they for?
```
const [open, setOpen] = useState(false);
const [save, setSave] = useState(false);
```

Reply:
Open is used for rendering the pop up. If set to True - pop up appears.

save on the other hand is used only when the Save button is clicked, then it triggers the PATCH method.